pkg/server/plugin/zmq/broker.go

```diff
@@ -135,110 +150,223 @@ func (lb *Broker) Run() {
 			}
 			if len(msg) == 1 {
 				status := string(msg[0])
-				lb.handleWorkerStatus(&lb.workers, address, status)
+				lb.handleWorkerStatus(address, status)
 			} else {
-				lb.frontend.SendMessage(msg)
+				lb.frontend <- msg
 				lb.logger.Debugf("zmq/broker: plugin => server: %#x, %s\n", msg[0], msg)
 			}
-		case lb.frontend:
-			frames, err := lb.frontend.RecvMessage()
+		case pull:
+			frames, err := pull.RecvMessage()
 			if err != nil {
 				panic(err)
 			}
-
-			frames = append([][]byte{lb.workers.Next()}, frames...)
-			lb.backend.SendMessage(frames)
-			lb.logger.Debugf("zmq/broker: server => plugin: %#x, %s\n", frames[0], frames)
+			backend.SendMessage(frames)
 		case nil:
```

Comment:
No. This switch case is only zmq socket. Not go chan. So we cannot consume the chan here.